File: Data Structures & Algorithms/longest-common-prefix/submission-1.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string prefix;
        string s1 = strs[0];
        cout << s1;
        bool yes = true;
        for(int i = 0; i < s1.length(); i++){
            for(string str : strs){
                if(i > str.length() || s1[i] != str[i]){
                    return prefix;
                }
            }
            prefix += s1[i];
        }
        return prefix;
    }
};

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.java
class Solution {
    public int subarraySum(int[] nums, int k) {
       int prefix = 0;
       int count = 0;
       Map<Integer, Integer> mp = new HashMap<>();
       mp.put(0, 1);
       for(int i : nums){
            prefix += i;
            int remove = prefix - k;
            if(mp.containsKey(remove)){
                count += mp.get(remove);
            }
            mp.put(prefix, mp.getOrDefault(prefix, 0)+1);
       }
       return count;
    }
}